.github/workflows/e2e.yml

```
@@ -30,7 +30,18 @@ jobs:
       - uses: pnpm/action-setup@c3b53f6a16e57305370b4ae5a540c2077a1d50dd # pin@v2.2.4
         with:
           version: 7
-      - run: cargo build --bin sui-test-validator --bin sui --profile dev
+
+      - name: Read Sui crate version
+        working-directory: ./
+        run: | 
+          echo "sui_version=$(cat Cargo.toml | grep '^version =' | tr -d '\"' | awk '{ print $3 }')" >> $GITHUB_ENV
+          echo "sui_version=${{ env.sui_version }}"
+
+      - name: Download sui-test-validator from current devnet release
+        uses: wei/wget@v1
+        with:
+          args: -O $PWD/target/debug/sui-test-validator https://github.com/MystenLabs/sui/releases/download/devnet-${{ env.sui_version }}/sui-test-validator
+
```

We should revert this particular change since here we do want to build a binary using the latest code on `main`
```suggestion
     - run: cargo build --bin sui-test-validator --bin sui --profile dev
```